### ISSUE TITLE
`semi`: `semi` rule doesn't support `accessor` property

### ISSUE THREAD

**Issue by dog-s-life** [eslint-stylistic/eslint-stylistic#777]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

`semi` rule doesn't work for `accessor` property.

```ts
class Foo {
    // this definition is allowed even `semi` rule is on
    accessor bar: string
    // expected
    accessor bar: string;
}
```

This would be fixed by adding `AST_NODE_TYPES.AccessorProperty` to `nodesToCheck`.

```diff
 // rules/semi.js
 // ...
     const nodesToCheck = [
+      AST_NODE_TYPES.AccessorProperty,
       AST_NODE_TYPES.PropertyDefinition,
       AST_NODE_TYPES.TSAbstractPropertyDefinition,
       AST_NODE_TYPES.TSDeclareFunction,
```

### Reproduction

https://stackblitz.com/edit/github-mk6sngpd-yhqqlchb

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
